curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/65ca2d0625aa3a3201067f70.md

```diff
@@ -0,0 +1,150 @@
+---
+id: 65ca2d0625aa3a3201067f70
+title: Step 30
+challengeType: 0
+dashedName: step-30
+---
+
+# --description--
+
+In the callback function of your `forEach`, use the `el` parameter to access the `counts` object. If the element is not in `counts`, set it to `0`. Otherwise, increment the count for the element. 
+
+Below your `forEach`, log out the final result of `counts`.
+
+# --hints--
+
+Your `.forEach()` method should increment the count for each element. Don't forget the initialization scenario. Enter numbers `4, 4, 2, 5` and click `Calculate` for testing.
+
+```js
+const expected = {'2': 1, '4': 2, '5': 1};
+assert.deepEqual(getMode([4, 4, 2, 5]), expected)
+```
+
+You should have a `console.log` for your `counts` variable.
+
+```js
+assert.match(getMode.toString(), /console\.log\(\s*counts\s*\)/);
```

Comment:
A new test should be added to make sure they changed it to return counts instead of return array.